Drivers/Src/CANbus.c

```diff
@@ -39,16 +39,29 @@ static void CANbus_CountIncoming(void) {
 /**
  * @brief   Initializes the CAN system
  * @param   loopback	: if we should use loopback mode (for testing)	
+ * @param 	faultState  : determines whether to implement Rx and Tx interrupts
  * @return  None
  */
-void CANbus_Init(bool loopback) {
-
-	RTOS_BPS_MutexCreate(&CANbus_TxMutex, "CAN TX Lock");
-	RTOS_BPS_MutexCreate(&CANbus_RxMutex, "CAN RX Lock");
-	RTOS_BPS_SemCreate(&CANbus_MailSem4, "CAN Mailbox Semaphore", 3); // # of mailboxes
-	RTOS_BPS_SemCreate(&CANbus_ReceiveSem4, "CAN Queue Counter Semaphore", 0);
-	// Initialize and pass interrupt hooks
-    BSP_CAN_Init(CANbus_CountIncoming, CANbus_Release, loopback);
+void CANbus_Init(bool loopback, bool faultState) {
```

Comment:
should add a note saying that you must call CANBus_DeInit() before CANbus_Init if you are reinitializing for Fault State.

Reply:
fixed

Reply:
You should use the `@NOTE` parameter. It's fine for now though